### PR TITLE
podmanv2 import

### PR DESCRIPTION
**Pull request by baude**
add the ability to import a container image from a container export

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: pkg/domain/infra/tunnel/images.go [hunk 1]
```diff
@@ -93,3 +94,20 @@ func (ir *ImageEngine) Pull(ctx context.Context, rawImage string, options entiti
...
+func (ir *ImageEngine) Import(ctx context.Context, opts entities.ImageImportOptions) (*entities.ImageImportReport, error) {
+	var (
+		err       error
+		sourceURL *string
+		f         *os.File
+	)
+	if opts.SourceIsURL {
+		sourceURL = &opts.Source
+	} else {
+		f, err = os.Open(opts.Source)
```
Comment (jwhonce) [inline pkg/domain/infra/tunnel/images.go]: This is importing off the server, not the client.  Won't the user expect the file to be on their machine?

File: cmd/podmanV2/images/import.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	}
+
+	importOpts.Source = source
+	importOpts.Reference = reference
+
+	response, err := registry.ImageEngine().Import(context.Background(), importOpts)
+	if err != nil {
+		return err
+	}
+	fmt.Println(response.Id)
```
Comment (vrothberg) [inline cmd/podmanV2/images/import.go:85]: Unrelated to this PR. I don't get why we're handcrafting return types in `entities`. Non of these types need to be marshaled, so we could very well just return a `string` in this case (and many others). No need to change now but this is just a general comment.

File: pkg/domain/infra/tunnel/images.go (hunk 1 above)
Comment (vrothberg) [reply pkg/domain/infra/tunnel/images.go:107]: We're still on the client here and are passing it to the bindings which eventually throw it over to the server.

File: cmd/podmanV2/images/import.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: importCommand,
+	})
+
+	importCommand.SetHelpTemplate(registry.HelpTemplate())
+	importCommand.SetUsageTemplate(registry.UsageTemplate())
+	flags := importCommand.Flags()
+	flags.StringArrayVarP(&importOpts.Changes, "change", "c", []string{}, "Apply the following possible instructions to the created image (default []): CMD | ENTRYPOINT | ENV | EXPOSE | LABEL | STOPSIGNAL | USER | VOLUME | WORKDIR")
+	flags.StringVarP(&importOpts.Message, "message", "m", "", "Set commit message for imported image")
+	flags.BoolVarP(&importOpts.Quiet, "quiet", "q", false, "Suppress output")
```
Comment (vrothberg) [inline cmd/podmanV2/images/import.go:47]: Long-term proposal: I'd love to have an `--input -i` as we have for `load`. We inherited it from Docker but can remain backward compat even when adding a new option.

File: pkg/domain/entities/images.go [hunk 4]
```diff
@@ -171,3 +171,16 @@ type ImagePruneReport struct {
 	Report Report
 	Size   int64
 }
+
+type ImageImportOptions struct {
```
Comment (vrothberg) [inline pkg/domain/entities/images.go:175]: Nit: if we comment all this stuff now, we don't need to do it in the future. Let's name it while it's hot?

File: pkg/domain/entities/images.go [hunk 5]
```diff
@@ -171,3 +171,16 @@ type ImagePruneReport struct {
...
+	Changes     []string
+	Message     string
+	Quiet       bool
+	Reference   string
+	Source      string
+	SourceIsURL bool
+}
+
+type ImageImportReport struct {
+	Id string
```
Comment (vrothberg) [inline pkg/domain/entities/images.go:185]: s/Id/ID/ ... for the linter

File: cmd/podmanV2/images/import.go [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+var (
+	importDescription = `Create a container image from the contents of the specified tarball (.tar, .tar.gz, .tgz, .bzip, .tar.xz, .txz).
+
+  Note remote tar balls can be specified, via web address.
+  Optionally tag the image. You can specify the instructions using the --change option.`
+	importCommand = &cobra.Command{
+		Use:               "import [flags] PATH [REFERENCE]",
+		Short:             "Import a tarball to create a filesystem image",
+		Long:              importDescription,
+		RunE:              importCon,
```
[the remark this answers is not in the document]
Comment (baude) [reply cmd/podmanV2/images/import.go:24]: i was trying to say importContainer :)

File: cmd/podmanV2/images/import.go (hunk 2 above)
Comment (baude) [reply cmd/podmanV2/images/import.go:85]: we agreed in principle that everything should feed a struct and return a struct for future wiggle room

File: pkg/domain/entities/images.go (hunk 5 above)
Comment (baude) [reply pkg/domain/entities/images.go:188]: we get into this push and pull between go templates and the linter here. leaving for now.

**Comment by vrothberg** [comment]
Making chief happy.
/lgtm
/hold

File: cmd/podmanV2/images/import.go [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+		PersistentPreRunE: preRunE,
+		Example: `podman import http://example.com/ctr.tar url-image
+  cat ctr.tar | podman -q import --message "importing the ctr.tar tarball" - image-imported
+  cat ctr.tar | podman import -`,
+	}
+)
+
```
[the remark this answers is not in the document]
Comment (vrothberg) [reply cmd/podmanV2/images/import.go:31]: Still outstanding.

**Pull request merged**